src/logic/synchronization.ts

```diff
@@ -87,11 +92,17 @@ export async function* startSynchronization(
     const exportPromises = _.keys(metadata)
         .map(type => {
             const myClass = d2ModelFactory(d2, type);
+
+            const metadataType = myClass.getMetadataType();
             return {
                 type,
                 ids: metadata[type].map(e => e.id),
-                excludeRules: myClass.getExcludeRules(),
-                includeRules: myClass.getIncludeRules(),
+                excludeRules: metadataIncludeExcludeRules
+                    ? metadataIncludeExcludeRules[metadataType].excludeRules.map(_.toPath)
+                    : myClass.getExcludeRules(),
+                includeRules: metadataIncludeExcludeRules
+                    ? metadataIncludeExcludeRules[metadataType].includeRules.map(_.toPath)
+                    : myClass.getIncludeRules(),
```

Comment:
I would use ``useDefaultIncludeExclude`` instead of undefined-checking here.

Reply:
The really important field to sync is metadataIncludeExcludeRules, the useDefaultIncludeExclude field is a helper field for when you are creating the sync rule.

Using useDefaultIncludeExclude field is not sufficient to decide using hardcoded rules in models or metadataIncludeExcludeRules because metadataIncludeExcludeRules can be undefined.

there two options:
- current 
       excludeRules: metadataIncludeExcludeRules
                    ? metadataIncludeExcludeRules[metadataType].excludeRules.map(_.toPath)
                    : myClass.getExcludeRules(),
                includeRules: metadataIncludeExcludeRules
                    ? metadataIncludeExcludeRules[metadataType].includeRules.map(_.toPath)
                    : myClass.getIncludeRules(),
- with useDefaultIncludeExclude 
           excludeRules: metadataIncludeExcludeRules && useDefaultIncludeExclude
                    ? metadataIncludeExcludeRules[metadataType].excludeRules.map(_.toPath)
                    : myClass.getExcludeRules(),
                includeRules: metadataIncludeExcludeRules && useDefaultIncludeExclude
                    ? metadataIncludeExcludeRules[metadataType].includeRules.map(_.toPath)
                    : myClass.getIncludeRules(),

I think the current option is more simpler.

Reply:
Done